internal/clients/team4/forage.go

```diff
@@ -16,15 +17,62 @@ type forageHistory struct {
 	numberCaught   uint
 }
 
-//func (c *client) analyseHistory() {
-//lookBack
-//}
+func (c *client) analyseHistory() {
+	constLookBack := 5
+	if c.getTurn() < 5 {
+		c.forage.preferedForageMethod = shared.DeerForageType
+		return
+	}
+
+	totalResources := make(map[shared.ForageType]float64)
+	totalResources[shared.DeerForageType] = 1.1
+	totalResources[shared.FishForageType] = 1
+
+	if len(c.forage.forageHistory) < constLookBack {
+		constLookBack = len(c.forage.forageHistory)
+	}
+	for _, e := range c.forage.forageHistory[len(c.forage.forageHistory)-constLookBack:] {
+		var ratio float64
+		if e.decision.Contribution <= 0 {
+			ratio = 1
+		} else {
+			ratio = float64(e.resourceReturn) / float64(e.decision.Contribution)
+		}
+		totalResources[e.decision.Type] = totalResources[e.decision.Type] * ratio
+	}
+
+	if len(c.forage.receivedForageData) < constLookBack {
+		constLookBack = len(c.forage.receivedForageData)
+	}
+	for _, e := range c.forage.receivedForageData[len(c.forage.receivedForageData)-constLookBack:] {
+		for _, teamEntry := range e {
+			var ratio float64
+			if teamEntry.DecisionMade.Contribution <= 0 {
+				ratio = 1
+			} else {
+				ratio = float64(teamEntry.ResourceObtained) / float64(teamEntry.DecisionMade.Contribution)
+			}
+			totalResources[teamEntry.DecisionMade.Type] = totalResources[teamEntry.DecisionMade.Type] * ratio
+		}
+		if totalResources[shared.DeerForageType] >= totalResources[shared.FishForageType] {
+			c.forage.preferedForageMethod = shared.DeerForageType
+		} else {
+			c.forage.preferedForageMethod = shared.FishForageType
+		}
+	}
+}
 
 func (c *client) DecideForage() (shared.ForageDecision, error) {
-	//analyseHistory()
+	c.analyseHistory()
```

Comment:
is there a reason why you have a void function that sets internal state (`c.forage` instead of returning the forage object from `c.analyseHistory()`

Reply:
is `c.forage` used anywhere else?